src/components/fields/image-upload/image-upload.tsx

```diff
@@ -60,11 +61,11 @@ export const ImageUploadInner = (props: IGenericFieldProps<IImageUploadSchema>)
 	// =============================================================================
 	useEffect(() => {
 		dispatchFieldEvent("mount", id);
-	}, []);
+	}, [dispatchFieldEvent, id]);
 
 	useEffect(() => {
```

Comment:
add `isTouched` to the dependencies, do the same for `file-upload`

Reply:
updated the dependency array